feat: add support for `React.use()` (#7988)
TanStack/query: docs/framework/react/reference/queryOptions.md

- `queryKey: QueryKey`
  - **Required**
  - The query key to generate options for.
- `experimental_prefetchInRender?: boolean`
  - Optional
  - Defaults to `false`

>   - When set to `true`, enables the experimental `useQuery().promise` functionality
>   - This allows prefetching queries during render, which can be useful for certain optimization scenarios

Reviewer: ```suggestion
  - When set to `true`, queries will be prefetched during render, which can be useful for certain optimization scenarios
  - Needs to be turned on for the experimental `useQuery().promise` functionality
```
